specs/capri/Multi-attach.md

```diff
@@ -0,0 +1,187 @@
+# Title
+
+**Author(s)**: [Jack Li](https://github.com/jackhaibo)
+
+## Summary
+
+This design is proposed to enable MultiAttach of volumes.
+
+## Motivation
+
+Currently a request to attach a volume that is already attached (in-use) will be rejected. There are situations where a cloud user may have deployed a clustered file-system and they would like the ability to attach a volume to multiple hosts/servers at the same time.
+
+### Goals
+
+This spec proposes a set of changes to enable and control the use of MultiAttach volumes from the OpenSDS side.
+
+## Design Details
+
+1. The drivers that support multi-attach ability need to report this in their capabilities through pool configuration file. The multi-attach volume creation request will select multi-attach enabled pool from pools which come from different drivers and many of them may not support this ability. 
+2. Attach a volume as one would normally do (Read Write access) to a host and then attach it to another host in Read Only mode or Read Write Mode. In order to implement this, add an item AttachMode to the volume attachment which has two values: RO (Read Only) and RW (Read Write), default is RW. 
+3. Add multiAttach bool to the volume model to enables/disables the ability of multi-attach. 
```

Comment:
enable/disable rather than enables/disables
```suggestion
3. Add multiAttach bool to the volume model to enable/disable the ability of multi-attach. 
```

Reply:
Done.